jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6140 @@
+	 */
+	/* true if machine is big endian, false otherwise. */
+	private static final boolean IS_BIG_ENDIAN;
+	/* true if addresses are aligned in memory, false otherwise. */
+	private static final boolean UNALIGNED_ACCESS;
+
+	/* Size of int in bytes. */
+	private static final int BYTES_IN_INT = 4;
+	/* Number of bits in a short. */
+	private static final int BITS_IN_SHORT = 16;
+	/* Number of bits in a byte. */
+	private static final int BITS_IN_BYTE = 8;
+
+	/* 32 bit mask (long) */
+	private static final long INT_MASK = 0xFFFFFFFFL;
+	/* 16 bit mask (long) */
+	private static final long SHORT_MASK = 0xFFFFL;
+	/* 8 bit mask (long) */
+	private static final long BYTE_MASK = 0xFFL;
+	/* 16 bit mask (int) */
+	private static final int SHORT_MASK_INT = (int) SHORT_MASK;
+	/* 8 bit mask (int) */
+	private static final int BYTE_MASK_INT = (int) BYTE_MASK;
+
+	/* Mask aligns offset to be int addressable (all but bottom two bits). */
+	private static final int INT_OFFSET_ALIGN_MASK = 0xFFFFFFFC;
+
+	/* 
+	 * For int shift instructions (ishr, iushr, ishl), only the low 5 bits of the
+	 * shift amount are considered.
+	 */
+	private static final int BC_SHIFT_INT_MASK = 0b11111;
+	
+	/* 
+	 * For long shift instructions (lshr, lushr, lshl) only the low 6 bits of the
+	 * shift amount are considered.
+	 */
+	private static final int BC_SHIFT_LONG_MASK = 0b111111;
+
+	/* Mask byte offset of an int. */
+	private static final long BYTE_OFFSET_MASK = 0b11L;
+	
+	static {
+		registerNatives();
+
+		theUnsafe = new Unsafe();
+
+		INVALID_FIELD_OFFSET = -1;
+		ARRAY_BOOLEAN_BASE_OFFSET = theUnsafe.arrayBaseOffset(boolean[].class);
+		ARRAY_BYTE_BASE_OFFSET = theUnsafe.arrayBaseOffset(byte[].class);
+		ARRAY_SHORT_BASE_OFFSET = theUnsafe.arrayBaseOffset(short[].class);
+		ARRAY_CHAR_BASE_OFFSET = theUnsafe.arrayBaseOffset(char[].class);
+		ARRAY_INT_BASE_OFFSET = theUnsafe.arrayBaseOffset(int[].class);
+		ARRAY_LONG_BASE_OFFSET = theUnsafe.arrayBaseOffset(long[].class);
+		ARRAY_FLOAT_BASE_OFFSET = theUnsafe.arrayBaseOffset(float[].class);
+		ARRAY_DOUBLE_BASE_OFFSET = theUnsafe.arrayBaseOffset(double[].class);
+		ARRAY_OBJECT_BASE_OFFSET = theUnsafe.arrayBaseOffset(Object[].class);
+		ARRAY_BOOLEAN_INDEX_SCALE = theUnsafe.arrayIndexScale(boolean[].class);
+		ARRAY_BYTE_INDEX_SCALE = theUnsafe.arrayIndexScale(byte[].class);
+		ARRAY_SHORT_INDEX_SCALE = theUnsafe.arrayIndexScale(short[].class);
+		ARRAY_CHAR_INDEX_SCALE = theUnsafe.arrayIndexScale(char[].class);
+		ARRAY_INT_INDEX_SCALE = theUnsafe.arrayIndexScale(int[].class);
+		ARRAY_LONG_INDEX_SCALE = theUnsafe.arrayIndexScale(long[].class);
+		ARRAY_FLOAT_INDEX_SCALE = theUnsafe.arrayIndexScale(float[].class);
+		ARRAY_DOUBLE_INDEX_SCALE = theUnsafe.arrayIndexScale(double[].class);
+		ARRAY_OBJECT_INDEX_SCALE = theUnsafe.arrayIndexScale(Object[].class);
+
+		ADDRESS_SIZE = theUnsafe.addressSize0();
+		IS_BIG_ENDIAN = theUnsafe.isBigEndian0();
+		UNALIGNED_ACCESS = theUnsafe.unalignedAccess0();
```

Comment:
The native unconditionally returns TRUE.  Better to hardcode that here and avoid a call.  Can always be changed if we support a platform that doesn't allow unaligned access in the future.